src/passes/RemoveUnusedBrs.cpp

```diff
@@ -129,11 +130,22 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
           flows.resize(size - skip);
         }
         // drop a nop at the end of a block, which prevents a value flowing
-        while (block->list.size() > 0 && block->list.back()->is<Nop>()) {
-          block->list.resize(block->list.size() - 1);
+        while (list.size() > 0 && list.back()->is<Nop>()) {
+          list.resize(list.size() - 1);
           self->anotherCycle = true;
         }
       }
+      // A value flowing is only valid if it is a value that the block actually
+      // flows out. If it is never reached, it does not flow out, and may be
+      // invalid to represent as such.
+      auto size = list.size();
+      for (Index i = 0; i < size; i++) {
+        if (i != size - 1 && list[i]->type == unreachable) {
```

Comment:
Why not check the last element as well?

Reply:
The last element is the value that may be flowing out of the block.

Reply:
And it may be unreachable, consider
```
(block
  (..)
  (return (i32.const 1))
)
```
That last element's value is what we want to flow out.

Reply:
So you'd want to flow out the `i32.const 1` even if the `return` was unreachable?